scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala

```diff
@@ -169,10 +152,16 @@ final case class ParquetAvroIO[T: ClassTag: Coder](path: String) extends ScioIO[
 }
 
 object ParquetAvroIO {
+  object ReadParam {
+    private[avro] val DefaultProjection = null
+    private[avro] val DefaultPredicate = null
+    private[avro] val DefaultConfiguration = new Configuration()
+  }
   final case class ReadParam[A: ClassTag, T: ClassTag] private (
-    projection: Schema,
-    predicate: FilterPredicate,
-    projectionFn: A => T
+    projectionFn: A => T,
+    projection: Schema = ReadParam.DefaultProjection,
+    predicate: FilterPredicate = ReadParam.DefaultPredicate,
+    conf: Configuration = ReadParam.DefaultConfiguration
```

Comment:
Not sure how many users are really using this module but this will a breaking change. I believe we have a few more PR's like this and it will hard to track down for v0.10.0. maybe we should try to mark these PR's with a `breaking-change` label?  

for this case we could have an `apply` method with the old order and `@deprecated`.

Reply:
Yeah this is only to accommodate `projectionFn` being the only "required" arg now. I can revert and work around it some other way to keep source compat.

Reply:
Actually this constructor is `private` so probably fine?

Reply:
right, to create it is through the companion default `apply` `ReadParam(...)`. That's why I suggested adding just another one with the old order and all is good in the hood 😄

Reply:
Which one? `object ReadParam` is new and doesn't have an `apply` method. `case class ReadParam`'s constructor has always been private.